### PR TITLE
ICU-21555 fix typos for D, E and F found in the repo

### PR DESCRIPTION
**Pull request by erik0686**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21555
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by daniel-ju — APPROVED** [review]
lgtm, thanks!

File: icu4c/source/tools/gencfu/gencfu.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 //   The input rule filew is are plain text files containing confusable character
```
Comment (jefgen) [inline icu4c/source/tools/gencfu/gencfu.cpp:24]:
```suggestion
//   The input rule files are plain text files containing confusable character
```

File: icu4j/main/classes/core/src/com/ibm/icu/text/DecimalFormatSymbols.java [hunk 2]
```diff
@@ -965,7 +965,7 @@ public void setMonetaryDecimalSeparatorString(String sep) {
         if (sep.length() == 1) {
             this.monetarySeparator = sep.charAt(0);
         } else {
-            // Use default decimap separator character as fallbacl
+            // Use default decimap separator character as fallback
```
Comment (jefgen) [inline icu4j/main/classes/core/src/com/ibm/icu/text/DecimalFormatSymbols.java:968]:
```suggestion
            // Use default decimal separator character as fallback
```

**Review by jefgen — COMMENTED** [review]
Thanks @erik0686!

There were a few minor D and F typos I noticed; I noted them as suggestions so you can probably try to commit them and then squash. :)

**Pull request merged**